### PR TITLE
Parse IPv6 addresses.

### PR DESCRIPTION
**Pull request by swankjesse**
This leans on InetAddress to do the work. It's not perfect, but it's
also not a lot of code. https://github.com/square/okhttp/issues/1613

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java [hunk 1]
```diff
@@ -731,12 +736,20 @@ private static String host(String input, int pos, int limit) {
       return idnDecoded;
     }
 
-    private static String decodeIpv6(String input, int pos, int limit) {
-      return input.substring(pos, limit); // TODO(jwilson) implement IPv6 decoding.
+    private static InetAddress decodeIpv6(String input, int pos, int limit) {
+      try {
+        return InetAddress.getByName(input.substring(pos, limit));
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java]: Move substring to call site? Unless there's some other value I'm not seeing (on mobile).

**Comment by JakeWharton** [comment]
LGTM. Comment is matter of taste and optional.

**Comment by swankjesse** [comment]
I'm gonna merge as-is. If we ever switch to the handrolled parser I made, we can save one allocation.

**Comment by swankjesse** [comment]
https://github.com/square/okhttp/issues/1486

**Pull request merged**